lib/Elevate/Database.pm

```diff
@@ -0,0 +1,74 @@
+package Elevate::Database;
+
+=encoding utf-8
+
+=head1 NAME
+
+Elevate::Database
+
+Helper/Utility logic for database related tasks.
+
+=cut
+
+use cPstrict;
+
+use Log::Log4perl qw(:easy);
+
+use Elevate::OS ();
+
+sub is_database_provided_by_cloudlinux ( $use_cache = 1 ) {
+    return 0 unless Elevate::OS::provides_mysql_governor();
+
+    if ($use_cache) {
+        my $cloudlinux_database_installed = cpev::read_stage_file( 'cloudlinux_database_installed', 0 );
+        return 1 if $cloudlinux_database_installed;
+    }
+
+    # Returns undef if database is not provided by cloudlinux
+    my ( $db_type, $db_version ) = Elevate::Database::get_db_info_if_provided_by_cloudlinux();
+
+    return 1 if $db_type && $db_version;
+    return 0;
+}
+
+sub get_db_info_if_provided_by_cloudlinux () {
+    my $installed = cpev::yum_list();
+
+    my $db_type;
+    my $db_version;
+    foreach my $repo ( keys %$installed ) {
```

Comment:
I might be missing something, but it seems to me like, rather than iterating through the list of all installed packages, we could check which package provides the mysqld binary.

ie:

```
rpm -qf /usr/sbin/mysqld
```

Reply:
Nice!  Verified that this will work and it simplified this code significantly.  Thank you @cpanel-hans